client/client.go

```diff
@@ -0,0 +1,309 @@
+	req.Header.Add("Content-Type", "application/json")
+	if c.config.Username != "" && c.config.Password != "" {
+		req.SetBasicAuth(c.config.Username, c.config.Password)
+	}
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return r, fmt.Errorf("client: error sending request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	buf, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return r, fmt.Errorf("client: error reading response: %w", err)
+	}
+
+	var ar api.Result
+	err = json.Unmarshal(buf, &ar)
+	if err != nil {
+		return r, fmt.Errorf("client: error unmarshaling response: %w", err)
+	}
+
+	if ar.Error != "" {
+		return r, fmt.Errorf("client: server error: %s", ar.Error)
+	}
+
+	if len(ar.Results) == 0 {
+		return r, fmt.Errorf("client: server error: %#v", ar)
+	}
+
+	return ar.Results[0], nil
+}
+
+// GetTransaction fetches the given transaction definition from server
+func (c *Client) GetTransaction(name string) (t transaction.Config, err error) {
+	ts, err := c.getTransactions(name)
+	if err != nil {
+		return t, err
+	}
+
+	if len(ts) == 0 {
+		return t, fmt.Errorf("client: no transaction with name %s found", name)
+	}
+
+	return ts[0], nil
+}
+
+// GetTransactions fetches all transactions from server
+func (c *Client) GetTransactions() (t []transaction.Config, err error) {
+	return c.getTransactions("")
+}
+
+func (c *Client) getTransactions(name string) (t []transaction.Config, err error) {
+
+	var req *http.Request
+	switch name != "" {
+	case true:
+		req, err = http.NewRequest(http.MethodGet, c.config.URL+api.EndpointTransaction+"/"+name, nil)
+	case false:
+		req, err = http.NewRequest(http.MethodGet, c.config.URL+api.EndpointTransaction, nil)
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	if c.config.Username != "" && c.config.Password != "" {
+		req.SetBasicAuth(c.config.Username, c.config.Password)
+	}
+
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("client: error sending request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	buf, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("client: error reading response: %w", err)
+	}
+
+	var ar api.Result
+	err = json.Unmarshal(buf, &ar)
+	if err != nil {
+		return nil, fmt.Errorf("client: error unmarshaling response: %w", err)
+	}
+
+	if ar.Error != "" {
+		return nil, fmt.Errorf("client: server error: %s", ar.Error)
+	}
+
+	return ar.Transactions, nil
+}
+
+// GetResult fetches the latest result for the given transaction from the server
+func (c *Client) GetResult(name string) (r transaction.Result, err error) {
+	rs, err := c.getResults(name)
+	if err != nil {
+		return r, err
+	}
+
+	if len(rs) == 0 {
+		return r, fmt.Errorf("client: no transaction with name %s found", name)
+	}
+
+	return rs[0], nil
+}
+
+// GetResults fetches the latest result for all transactions from the server
+func (c *Client) GetResults() (t []transaction.Result, err error) {
+	return c.getResults("")
```

Comment:
This is kinda ugly. Making use of an empty string to get all instead of one.

Reply:
So what do you suggest as an alternative?

Reply:
In these scenarios it's better to provide the alternative code, IMO

Reply:
I think that it's totally fine for private receivers, otherwise you would have to add control flow statements everywhere assuming that you had specific private methods of each usecase which is to cumbersome.

Reply:
I can take that

Reply:
I can't resolve the issue, please do so @flaviofernandes004

Reply:
👍

Reply:
It's not pretty i agree. But being private doesn't hurt that much.